src/app/rundown/components/segment/segment.component.ts

```diff
@@ -1,25 +1,124 @@
-import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core'
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output,
+  SimpleChange,
+  SimpleChanges
+} from '@angular/core'
 import { Segment } from '../../../core/models/segment'
+import { Part } from '../../../core/models/part'
+import {PieceLayerService} from "../../../shared/services/piece-layer.service";
 
 @Component({
   selector: 'sofie-segment',
   templateUrl: './segment.component.html',
   styleUrls: ['./segment.component.scss']
 })
-export class SegmentComponent {
+export class SegmentComponent implements OnChanges, OnDestroy {
 
   @Input()
   public isRundownActive: boolean
 
+  @Input()
+  public isOnAir: boolean
+
   @Input()
   public segment: Segment
 
   @Output()
   setNextEvent: EventEmitter<{segmentId: string, partId: string}> = new EventEmitter()
 
-  constructor() { }
+  public timeReference: number = 0
+
+  public pieceLayers: string[] = []
+
+  private animationFrameId?: number
+
+  constructor(private readonly pieceLayerService: PieceLayerService) {}
 
   public emitSetNextEvent(partId: string): void {
     this.setNextEvent.emit({ segmentId: this.segment.id, partId})
   }
+
+  private getPieceLayers(): string[] {
+    const usedPieceLayers = this.pieceLayerService.getPieceLayersForParts(this.segment.parts)
+    const availablePieceLayersInOrder: string [] = [
+      'OVERLAY',
+      'PGM',
+      'JINGLE',
+      'MUSIK',
+      'MANUS',
+      'ADLIB',
+      'SEC',
+      'AUX',
```

Comment:
Since we currently don't plan on making these configurable, we should move them to a const or an enum.

Reply:
I vote for an Enum :)

Reply:
This has been extracted out into an enum and a method for setting the order.